test/cartesian/XAxis.spec.tsx

```diff
@@ -201,8 +324,146 @@ describe('<XAxis />', () => {
 
     const bar = container.querySelector('.recharts-rectangle');
     assertNotNull(bar);
-    expect(parseInt(bar.getAttribute('x') as string, 10)).toEqual(66);
-    expectXAxisTicks(container, ['100', '120', '140', '170']);
+    expect(bar.getAttribute('x')).toEqual('66.2928279883382');
+    expectXAxisTicks(container, [
+      {
+        textContent: '100',
+        x: '77.95510204081633',
+        y: '273',
+      },
+      {
+        textContent: '120',
+        x: '136.266472303207',
+        y: '273',
+      },
+      {
+        textContent: '140',
+        x: '194.57784256559765',
+        y: '273',
+      },
+      {
+        textContent: '170',
+        x: '282.0448979591837',
+        y: '273',
+      },
+    ]);
+    expect(spy).toHaveBeenLastCalledWith([100, 170]);
+  });
+
+  it('Render Bars with custom gap', () => {
```

Comment:
Here are the three new tests so you don't have to scroll through the ocean of `expectXAxisTicks`.